### PR TITLE
Randomize URLs in examples

### PR DESCRIPTION
**Pull request by bifurcation**
This PR updates the examples in the document to use the randomized URL scheme recommended in #445, in order to make clear the guidance to implementers.

Depends on #454

### CODE REVIEW

File: draft-ietf-acme-acme.md [hunk 1]
```diff
@@ -2497,7 +2515,7 @@ on randomness requirements.
 ~~~~~~~~~~
 {
   "type": "http-01",
-  "url": "https://example.com/acme/authz/0",
+  "url": "https://example.com/acme/authz/eYL3Xw",
```
Comment (felixfontein) [inline draft-ietf-acme-acme.md]: Is there a reason why `eYL3Xw` is much sohrter than the other authz identifiers used in the draft (`PAniVnsZcis`, `r4HqLzrSrpI`)?

File: draft-ietf-acme-acme.md [hunk 2]
```diff
@@ -1995,9 +1995,21 @@ paths to various trust anchors. Clients can fetch these alternates and use their
...
-Accept: application/pem-certificate-chain
+Content-Type: application/jose+json
+Accept: application/pkix-cert
+
+{
+  "protected": base64url({
+    "alg": "ES256",
+    "kid": "https://example.com/acme/acct/evOfKhNU60wg",
+    "nonce": "uQpSjlRb4vQVCjVYAyyUWg",
+    "url": "https://example.com/acme/cert/mAt3xBGaobw",
```
Comment (felixfontein) [inline draft-ietf-acme-acme.md]: The trailing comma is invalid JSON.

File: draft-ietf-acme-acme.md [hunk 3]
```diff
@@ -2059,17 +2077,17 @@ will have already received the pending authorization object in the response
...
 Accept: application/pkix-cert
 
 {
   "protected": base64url({
     "alg": "ES256",
-    "kid": "https://example.com/acme/acct/1",
+    "kid": "https://example.com/acme/acct/evOfKhNU60wg",
     "nonce": "uQpSjlRb4vQVCjVYAyyUWg",
-    "url": "https://example.com/acme/authz/1234",
+    "url": "https://example.com/acme/authz/PAniVnsZcis",
```
Comment (felixfontein) [inline draft-ietf-acme-acme.md]: The trailing comma is invalid JSON.

File: draft-ietf-acme-acme.md [hunk 4]
```diff
@@ -2169,17 +2187,17 @@ progress, the server MUST return a 200 (OK) response and MAY include a
...
```
Comment (felixfontein) [inline draft-ietf-acme-acme.md]: The trailing comma is invalid JSON.

File: draft-ietf-acme-acme.md (hunk 1 above)
Comment (bifurcation) [reply draft-ietf-acme-acme.md]: Search/replace error.  That should have been a challenge URL as well.

**Comment by jsha** [comment]
I oppose this PR, because I oppose the advice added in #445 to use capability URLs, since it dilutes our authentication model unnecessarily. Sorry I missed that during the review process; I'll follow up with a more detailed post on-list about removing it.

**Comment by bifurcation** [comment]
@jsha If we need to debate this in depth, we should do it on the list, but in case we can dispatch this quickly: I'm puzzled by your opposition to capability URLs.  When they're applied to POST-only URLs, they're strictly an improvement on the authentication / authz model.  It's only when they're applied to URLs where GETs are possible where there's some arguable weakening, in that you could choose GET+capability instead of POST+signature.  But if that's the case, then it seems like your issue is with allowing GETs at all, since once you allow GET, GET+capability is better than plain GET.

**Comment by bifurcation** [comment]
Actually, @jsha, I'm going to go ahead and merge this, because it doesn't actually conflict with your proposed change in #457

**Comment by jsha** [comment]
Actually, it does: The point is to not give people a false impression that URLs are unguessable when that is not a formal part of the security model. If it *is* a formal part of the security model, we need to analyze it properly.

**Pull request merged**